include/boost/graph/isomorphism.hpp

```diff
@@ -158,46 +174,65 @@ namespace detail
             );
         }
 
+        // Generates map of invariant multiplicity from sorted invariants
+        multiplicity_map multiplicities(const std::vector< invariant_t >& invariants) {
+          // Assumes invariants are sorted
+          multiplicity_map invar_multiplicity;
+
+          typedef typename std::vector< invariant_t >::const_iterator invar_iter;
+          typedef typename multiplicity_map::iterator invar_map_iter;
+          invar_iter it = invariants.begin();
+          const invar_iter end = invariants.end();
+
+          if(it == end) {
+            return invar_multiplicity;
+          }
+
+          invariant_t invar = *it;
+          invar_map_iter inserted = invar_multiplicity.emplace(invar, 1).first;
+          ++it;
+          for(; it != end; ++it)
+          {
+              if(*it == invar)
+              {
+                  inserted->second += 1;
+              }
+              else
+              {
+                  invar = *it;
+                  inserted = invar_multiplicity.emplace(invar, 1).first;
+              }
+          }
+
+          return invar_multiplicity;
```

Comment:
So wait, this looks like it can be simplified to just call `invar_multiplicity[invar]++`, because if `invar` doesn't already exist in the map then `operator[]` creates it. Which means this function boils down to `std::for_each` with a pretty simple lambda like `[](auto invariant){ invariant_map[invariant]++; }`. 
Now, since Boost officially deprecated C++03, I think you can feel free to use C++14 if that simplifies the implementation, but don't do any mass conversion from C++03 to C++14.

Reply:
It's not exactly the same! So the sorted list of invariants is laid out like e.g. `2, 2, 2, 7, 7, 9, 13` etc., right? The proposed version above doesn't repeatedly find the key position in the map for equal successive invariants, while a `for_each` version does. I assume it makes more of a performance difference the more repeated values there are. The default vertex invariant is the vertex degree, of which there are probably many equal values in large graphs. I haven't benchmarked anything, but I'm perfectly willing to.

Reply:
Ah, no, you're right, it would be silly to do it the way I suggested.